pkg/xds/envoy/listeners/v3/retry_configurer.go

```diff
@@ -93,6 +93,21 @@ func genHttpRetryPolicy(
 		policy.RetriableStatusCodes = conf.RetriableStatusCodes
 	}
 
+	for _, methodValue := range conf.GetRetriableMethods() {
+		if methodValue == mesh_proto.HttpMethod_NONE {
+			continue
+		}
+
+		method := mesh_proto.HttpMethod_name[int32(methodValue)]
+
+		policy.RetriableRequestHeaders = append(policy.RetriableRequestHeaders,
+			&envoy_route.HeaderMatcher{
+				Name:                 ":method",
+				HeaderMatchSpecifier: &envoy_route.HeaderMatcher_ExactMatch{ExactMatch: method},
```

Comment:
```suggestion
HeaderMatchSpecifier: &envoy_route.HeaderMatcher_ExactMatch{
        ExactMatch: m.String(),
 },
```

protobuf will implement [stringer](https://pkg.go.dev/fmt#Stringer) for enum types, so you don't need to manually index the names array.